src/frontend/screens/Settings/MapManagement/BackgroundMaps.tsx

```diff
@@ -0,0 +1,336 @@
+  const {formatMessage: t} = useIntl();
+
+  const mapAddedBottomSheet = useBottomSheetModal({openOnMount: false});
+  const removeMapBottomSheet = useBottomSheetModal({openOnMount: false});
+
+  const selectFileMutation = useSelectFile();
+  const importCustomMapMutation = useImportCustomMapFile();
+  const removeCustomMapMutation = useRemoveCustomMapFile();
+  const customMapInfoQuery = useGetCustomMapInfo();
+
+  return (
+    <>
+      <ScrollView contentContainerStyle={styles.container}>
+        <Text style={styles.aboutText}>{t(m.about)}</Text>
+        <View style={styles.descriptionContainer}>
+          <Text>{t(m.description1)}</Text>
+          <Text>{t(m.description2)}</Text>
+        </View>
+
+        <CustomMapInfoSection
+          onChooseFile={() => {
+            selectFileMutation.mutate(
+              {
+                extensionFilters: ['smp'],
+              },
+              {
+                onSuccess: asset => {
+                  if (!asset) return;
+
+                  importCustomMapMutation.mutate(
+                    {
+                      uri: asset.uri,
+                    },
+                    {
+                      onError: () => {
+                        FileSystem.deleteAsync(asset.uri, {
+                          idempotent: true,
+                        }).catch(noop);
+                      },
+                      onSuccess: () => {
+                        mapAddedBottomSheet.openSheet();
+                      },
+                    },
+                  );
+                },
+              },
+            );
+          }}
+          onRemoveMap={() => {
+            removeMapBottomSheet.openSheet();
+          }}
+        />
+
+        {customMapInfoQuery.status === 'error' && (
+          <>
+            <Text style={styles.infoLoadErrorText}>
+              {t(m.customMapInfoLoadError)}
+            </Text>
+            <Button
+              fullWidth
+              variant="outlined"
+              onPress={() => {
+                removeCustomMapMutation.mutate();
+              }}>
+              <Text style={styles.removeMapFileButtonText}>
+                {t(m.removeMapFile)}
+              </Text>
+            </Button>
+          </>
+        )}
+      </ScrollView>
+
+      <BottomSheetModal
+        ref={removeMapBottomSheet.sheetRef}
+        isOpen={removeMapBottomSheet.isOpen}>
+        <BottomSheetModalContent
+          icon={<ErrorSvg />}
+          title={t(m.deleteCustomMapTitle)}
+          description={
+            t(m.deleteCustomMapDescription) + '\n\n' + t(m.cannotBeUndone)
+          }
+          buttonConfigs={[
+            {
+              dangerous: true,
+              variation: 'filled',
+              text: t(m.deleteMapButtonText),
+              icon: <MaterialIcon size={30} name="delete" color={WHITE} />,
+              onPress: () => {
+                removeCustomMapMutation.mutate();
+                removeMapBottomSheet.closeSheet();
```

Comment:
not super concerned because this happens so quickly and also we have a different modal for an error that may occur. Added a small change anyways just in case though, via b79e78d5f8d165a156f4fa8cd265f8e708329875